lms/djangoapps/mobile_api/users/views.py

```diff
@@ -341,47 +362,147 @@ def get_serializer_class(self):
             return CourseEnrollmentSerializerv05
         return CourseEnrollmentSerializer
 
-    def get_queryset(self):
+    @cached_property
+    def queryset(self):
```

Comment:
Yes, you are right. For the first time we try to put all logic into `get_queryset `, but we received more then 400 SQL queries for the one request (!). So, we need to use property method for received `queryset`. Caching helps maximize performance, so we need to `@cached_property` decorator to make this API faster. Now we have about 120 SQL queries for the one API request. This is the maximum number of SQL queries when requesting additional fields and a large number of the user's enrollments.

Reply:
This [stackflow answer](https://stackoverflow.com/questions/19707237/use-get-queryset-method-or-set-queryset-variable/19707855#19707855) the best example of when to use queryset and get_queryset.
In our case we should move all request level logic to get_queryset and use caching there. queryset is already cached since it is evaluated only once.